nixos/tests/exporters.nix

```diff
@@ -0,0 +1,75 @@
+import ./make-test.nix ({ lib, ... }:
+{
+  name = "exporters";
+
+  meta = with lib.maintainers; {
+    maintainers = [ willibutz ];
+  };
+
+  machine = { config, pkgs, ... }: {
+    services.nixosManual.enable = lib.mkForce true;
+    services.prometheus.exporters = {
+      blackbox = {
+        enable = true;
+        openFirewall = true;
+        configFile = "/dev/null";
+      };
+      collectd = {
+        enable = true;
+      };
+      fritzbox = {
+        enable = true;
+      };
+      json = {
+        enable = true;
+        url = "localhost";
+        configFile = "/dev/null";
+      };
+      minio = {
+        enable = true;
+        minioAddress = "https://localhost:9000";
+        minioAccessKey = "lurmlurm";
+        minioAccessSecret = "lurmlurm";
+      };
+      nginx = {
+        enable = true;
+      };
+      node = {
+        enable = true;
+      };
+      snmp = {
+        enable = true;
+      };
+      unifi = {
+        enable = true;
+        unifiAddress = "http://localhost:8443";
+        unifiUsername = "fnord";
+        unifiPassword = "lurmlurm";
+      };
+      varnish = {
+        enable = true;
+      };
+    };
+  };
+
+  testScript = with lib; ''
+    $machine->start;
+    $machine->waitForUnit("multi-user.target");
+  '' + (concatStringsSep "\n" (flip map [
+    "blackbox" # FIXME
+    "collectd"
+    "fritzbox"
+    "json"
+    "minio"
+    "nginx" # FIXME
+    "node"
+    "snmp"
+    # "unifi" # FIXME
+    # "varnish" # FIXME
```

Comment:
What is wrong with these FIXMEs?

Reply:
Oh no, I forgot to take out the test %)
I was just using it for myself.
(These exporters fail because they can't scrape anything in this VM)